code/game/objects/structures/barrel.dm

```diff
@@ -1,15 +1,52 @@
 /obj/structure/reagent_dispensers/barrel
-	name                = "barrel"
-	desc                = "A stout barrel for storing large amounts of liquids or substances."
-	icon                = 'icons/obj/structures/barrel.dmi'
-	icon_state          = ICON_STATE_WORLD
-	anchored            = TRUE
-	atom_flags          = ATOM_FLAG_CLIMBABLE | ATOM_FLAG_OPEN_CONTAINER
-	matter              = null
-	material            = /decl/material/solid/organic/wood
-	material_alteration = MAT_FLAG_ALTERATION_COLOR | MAT_FLAG_ALTERATION_NAME | MAT_FLAG_ALTERATION_DESC
-	wrenchable          = FALSE
-	storage             = /datum/storage/hopper/industrial
+	name                      = "barrel"
+	desc                      = "A stout barrel for storing large amounts of liquids or substances."
+	icon                      = 'icons/obj/structures/barrel.dmi'
+	icon_state                = ICON_STATE_WORLD
+	anchored                  = TRUE
+	atom_flags                = ATOM_FLAG_CLIMBABLE | ATOM_FLAG_OPEN_CONTAINER
+	matter                    = null
+	material                  = /decl/material/solid/organic/wood
+	material_alteration       = MAT_FLAG_ALTERATION_COLOR | MAT_FLAG_ALTERATION_NAME | MAT_FLAG_ALTERATION_DESC
+	wrenchable                = FALSE
+	storage                   = /datum/storage/barrel
+	amount_dispensed          = 10
+	possible_transfer_amounts = @"[10,25,50,100]"
+	volume                    = 7500
+	atom_flags                = ATOM_FLAG_CLIMBABLE
+	movable_flags             = MOVABLE_FLAG_WHEELED
+
+/obj/structure/reagent_dispensers/barrel/Initialize()
+	..()
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/structure/reagent_dispensers/barrel/LateInitialize()
+	..()
+	for(var/obj/item/thing in loc)
+		if(!thing.simulated || thing.anchored)
+			continue
+		if(storage.can_be_inserted(thing, null))
+			storage.handle_item_insertion(null, thing)
```

Comment:
this results in `/datum/unit_test/chemistry_premade_bottles_shall_not_melt` failing because the loc of the spawned stuff is no longer the turf. maybe this should be a mapload only thing?